### PR TITLE
[scripts] MATN-TC-12: Hop limit processing

### PR DESCRIPTION
**Pull request by superwhd**
- Implement test case MATN-TC-12
- Also implemented a fake layer: 'ipv6inner'. It is useful for fetching data in the inner IPv6 layer from a packet.

### CODE REVIEW

**Review by simonlingoogle — COMMENTED** [review]
SGTM overall. 

Thanks for enhance `pktverify` to support `ipv6inner.hlim`, which is a feature we wanted for a long time.

File: tests/scripts/thread-cert/pktverify/layer_fields.py [hunk 1]
```diff
@@ -749,6 +758,6 @@ def _get_candidate_layers(packet, layer_name):
     layers = []
     for ln in candidate_layer_names:
         if hasattr(packet, ln):
-            layers.append(getattr(packet, ln))
+            layers.append(packet.get_multiple_layers(ln))
```
[the remark this answers is not in the document]
Comment (superwhd) [reply tests/scripts/thread-cert/pktverify/layer_fields.py:761]: After considering it for a while, I'd still keep it still. It's necessary for `_get_candidate_layers()` to return list of list for `ipv6inner`. If it returns a flattened list, we may not be able to fetch the wanted element using the index.

[the remark this answers is not in the document]
Comment (simonlingoogle) [reply tests/scripts/thread-cert/pktverify/layer_fields.py:761]: SGTM

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**